client/packages/system/src/Manage/Preferences/api/useEditPreference.ts

```diff
@@ -1,27 +1,47 @@
-import { FnUtils, isEmpty, useMutation } from '@openmsupply-client/common';
-import { JsonData } from '@openmsupply-client/programs';
+import {
+  isEmpty,
+  PreferenceNodeType,
+  UpsertPreferencesInput,
+  useAuthApi,
+  useMutation,
+  useNotification,
+  useTranslation,
+} from '@openmsupply-client/common';
 import { usePreferencesGraphQL } from './usePreferencesGraphQL';
+import { PREFERENCE_DESCRIPTIONS } from './keys';
+import { useAdminPrefsList } from './useAdminPrefsList';
 
-export const useEditPreference = (key: string) => {
+export const useEditPreferences = (prefType: PreferenceNodeType) => {
+  const t = useTranslation();
+  const { error } = useNotification();
+
+  const { data } = useAdminPrefsList(prefType);
+  const { mutateAsync } = useUpsertPref();
+
+  // Probably worth debouncing at some stage :)
```

Comment:
>  // Probably worth debouncing at some stage :)

Indeed -- definitely once we get a string input pref. Maybe update comment to "Please add debouncing once string prefs are implemented"?

Reply:
ah yeah nice shout